packages/dds/test-dds-utils/src/ddsFuzzHarness.ts

```diff
@@ -242,6 +284,72 @@ const defaultDDSFuzzSuiteOptions: DDSFuzzSuiteOptions = {
 	validationStrategy: { type: "random", probability: 0.05 },
 };
 
+function mixinNewClient<
+	TChannelFactory extends IChannelFactory,
+	TOperation extends BaseOperation,
+	TState extends DDSFuzzTestState<TChannelFactory>,
+>(
+	model: DDSFuzzModel<TChannelFactory, TOperation, TState>,
+	options: DDSFuzzSuiteOptions,
+): DDSFuzzModel<TChannelFactory, TOperation | AddClient, TState> {
+	const isClientAddOp = (op: TOperation | AddClient): op is AddClient => op.type === "addClient";
+
+	const generatorFactory: () => Generator<TOperation | AddClient, TState> = () => {
+		const baseGenerator = model.generatorFactory();
+		return async (state): Promise<TOperation | AddClient | typeof done> => {
+			const baseOp = baseGenerator(state);
+			const { clients, random } = state;
+			if (
+				options.clientJoinOptions !== undefined &&
+				clients.length < options.clientJoinOptions.maxNumberOfClients &&
+				random.bool(options.clientJoinOptions.clientAddProbability)
+			) {
+				return {
+					type: "addClient",
+					addedChannelId: makeFriendlyClientId(random, clients.length),
+				};
+			}
+			return baseOp;
+		};
+	};
+
+	const reducer: Reducer<TOperation | AddClient, TState> = async (state, op) => {
+		if (isClientAddOp(op)) {
+			const { summary } = state.summarizerClient.channel.getAttachSummary();
+			const dataStoreRuntime = new MockFluidDataStoreRuntime();
+			const containerRuntime = state.containerRuntimeFactory.createContainerRuntime(
```

Comment:
This code seems to depend on very specific behavior from our shared mocks, specifically, that when a new MockContainerRuntime is created, it will add itself to the quorum of the factory that created it, so the quorum state matches the number of clients. It took a bit to figure out how the quorum was being maintained here. I have mixed feelings about reusing the shared mocks, but i think adding unit test in the package that validates that behavior holds should be added, as i could see it being modified to support read by default, and some transition to write, which may then need updates in these tests which may be hard to identify

Reply:
done. there weren't any tests in this package before. if you want I can split off the infra boilerplate from here.